packages/storage/src/postgres.rs

```diff
@@ -678,6 +685,134 @@ impl Postgres {
         Ok(())
     }
 
+    pub(crate) async fn _update_costs(
+        &self,
+        cost_per_tx: Vec<([u8; 32], u128, u64)>,
+    ) -> Result<()> {
```

Comment:
does this have to be batched? The transactions seem to not care about each other?

If not, it might be better to do it tx by tx so that we don't hold the db tx for longer than we need to. No real performance impact since we aren't doing batch queries anyway.